test/distributed/fsdp/test_fsdp_optim_state.py

```diff
@@ -477,12 +480,21 @@ def test_full_optim_state_dict_keys(self):
         device = torch.device("cuda")
         model = NestedModel().to(device)
         wrapped_model = NestedModel.wrap(model, ignore_modules=True)
+        # Add checkpointing to ensure optim_state_dict and state_dict strip out
+        # checkpointing prefixes.
+        apply_activation_checkpointing_wrapper(
+            model,
+            check_fn=lambda module: isinstance(module, torch.nn.Sequential)
+        )
         optim = torch.optim.Adam(wrapped_model.parameters(), lr=1e-3)
         self._step_model(model, optim, device)
         optim_state_dict = FSDP.full_optim_state_dict(wrapped_model, optim, rank0_only=False)
         with FSDP.state_dict_type(wrapped_model, StateDictType.FULL_STATE_DICT):
             state_dict = wrapped_model.state_dict()
         self.assertEqual(optim_state_dict["state"].keys(), state_dict.keys())
+        # Check that checkpointing prefix was indeed stripped.
+        for key in optim_state_dict["state"].keys():
+            self.assertFalse(_CHECKPOINT_PREFIX in key)
```

Comment:
nit: 😄 
```suggestion
            self.assertNotIn(_CHECKPOINT_PREFIX, key)
```
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNotIn